src/controller.coffee

```diff
@@ -4012,6 +4012,11 @@ Editor::viewportDimensions = ->
     height: @mainCanvas.height
   }
 
+# LINE LOCATION API
+# =================
+Editor::getLineBoundingBox = (row) ->
```

Comment:
maybe rename to getLineDrawingMetrics

and add things like distance-to-text baseline.